junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/ScriptExecutionConditionTests.java

```diff
@@ -143,23 +83,62 @@ void annotationDefaultValues(TestInfo info) {
 	}
 
-	private Script script(Type type, String... lines) {
-		return new Script( //
-			type, //
-			"Mock for " + type, //
-			Script.DEFAULT_SCRIPT_ENGINE_NAME, //
-			String.join("\n", lines), //
-			Script.DEFAULT_SCRIPT_REASON_PATTERN //
-		);
+	@Test
+	void throwingEvaluatorExceptionMessage() {
+		String message = "Mock for message";
+		ReflectiveOperationException cause = new ReflectiveOperationException("Mock for ReflectiveOperationException");
+		ScriptExecutionCondition.Evaluator evaluator = new ScriptExecutionCondition.ThrowingEvaluator(message, cause);
+		Exception e = assertThrows(Exception.class, () -> evaluator.evaluate(null, null));
+		assertTrue(e instanceof ExtensionConfigurationException);
+		assertEquals(message, e.getMessage());
+	}
+
+	@Test
+	void enabledDueToAnnotatedElementNotPresent() {
+		ScriptExecutionCondition condition = new ScriptExecutionCondition();
+		ExtensionContext context = Mockito.mock(ExtensionContext.class);
+		ConditionEvaluationResult result = condition.evaluateExecutionCondition(context);
+		assertFalse(result.isDisabled());
+		assertTrue(result.getReason().isPresent());
+		result.getReason().ifPresent(reason -> assertEquals("AnnotatedElement not present", reason));
 	}
 
-	private Bindings createDefaultContextBindings() {
-		Bindings bindings = new SimpleBindings();
-		bindings.put(Script.BIND_JUNIT_TAGS, Collections.emptySet());
-		bindings.put(Script.BIND_JUNIT_UNIQUE_ID, "Mock for UniqueId");
-		bindings.put(Script.BIND_JUNIT_DISPLAY_NAME, "Mock for DisplayName");
-		bindings.put(Script.BIND_JUNIT_CONFIGURATION_PARAMETER, Collections.emptyMap());
-		return bindings;
+	@Test
+	void enabledDueToAnnotationNotPresent() {
+		ScriptExecutionCondition condition = new ScriptExecutionCondition();
+		ExtensionContext context = Mockito.mock(ExtensionContext.class);
+		Optional<AnnotatedElement> optionalElement = Optional.of(ScriptExecutionConditionTests.class);
+		Mockito.when(context.getElement()).thenReturn(optionalElement);
+		ConditionEvaluationResult result = condition.evaluateExecutionCondition(context);
+		assertFalse(result.isDisabled());
+		assertTrue(result.getReason().isPresent());
+		result.getReason().ifPresent(reason -> assertEquals("Annotation not present", reason));
+	}
+
+	@Test
+	void throwingEvaluatorIsCreatedWhenScriptEngineIsNotAvailable() {
+		String nameOfScriptEngine = "javax.script.DoesNotExist";
+		String name = "org.junit.jupiter.engine.extension.ScriptExecutionEvaluator";
+		ScriptExecutionCondition.Evaluator evaluator = ScriptExecutionCondition.Evaluator.forName(nameOfScriptEngine,
+			name);
+		Exception e = assertThrows(Exception.class, () -> evaluator.evaluate(null, null));
+		assertTrue(e instanceof ExtensionConfigurationException);
+		String message = e.getMessage();
+		System.out.println(message);
+		assertTrue(message.startsWith("Class `" + nameOfScriptEngine + "` is not loadable"));
+		assertTrue(message.endsWith("`--add-modules ...,java.scripting`"));
```

Comment:
```java
assertThat(e).isInstanceOf(ExtensionConfigurationException.class)
		.hasMessageStartingWith("Class `" + nameOfScriptEngine + "` is not loadable")
		.hasMessageEndingWith("`--add-modules ...,java.scripting`")
```

Reply:
I like the one assertion per statement style. Will change it. :)